processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -218,6 +221,73 @@ public String getType()
     return GROUP_BY;
   }
 
+  @Override
+  public Ordering getResultOrdering()
+  {
+    final Comparator naturalNullsFirst = Ordering.natural().nullsFirst();
+    final Ordering<Row> rowOrdering = getRowOrdering(false);
+
+    return Ordering.from(
+        new Comparator<Object>()
+        {
+          @Override
+          public int compare(Object lhs, Object rhs)
+          {
+            if (lhs instanceof Row) {
+              return rowOrdering.compare((Row) lhs, (Row) rhs);
+            } else {
+              // Probably bySegment queries
+              return naturalNullsFirst.compare(lhs, rhs);
+            }
+          }
+        }
+    );
+  }
+
+  public Ordering<Row> getRowOrdering(final boolean granular)
+  {
+    final Comparator naturalNullsFirst = Ordering.natural().nullsFirst();
```

Comment:
I tried this and the benchmarks didn't change at all, so am leaning towards keeping it the way it is as the code is simpler.